homeassistant/components/climate/evohome.py

```diff
@@ -0,0 +1,395 @@
+EVO_RESET = 'AutoWithReset'
+EVO_AUTO = 'Auto'
+EVO_AUTOECO = 'AutoWithEco'
+EVO_AWAY = 'Away'
+EVO_DAYOFF = 'DayOff'
+EVO_CUSTOM = 'Custom'
+EVO_HEATOFF = 'HeatingOff'
+
+TCS_OP_LIST = [
+    STATE_AUTO,
+    STATE_ECO,
+    STATE_OFF
+]
+
+EVO_STATE_TO_HA = {
+    EVO_RESET: STATE_AUTO,
+    EVO_AUTO: STATE_AUTO,
+    EVO_AUTOECO: STATE_ECO,
+    EVO_AWAY: STATE_AUTO,
+    EVO_DAYOFF: STATE_AUTO,
+    EVO_CUSTOM: STATE_AUTO,
+    EVO_HEATOFF: STATE_OFF
+}
+
+HA_STATE_TO_EVO = {
+    STATE_AUTO: EVO_AUTO,
+    STATE_ECO: EVO_AUTOECO,
+    STATE_OFF: EVO_HEATOFF
+}
+
+# these are used to help prevent E501 (line too long) violations
+GWS = 'gateways'
+TCS = 'temperatureControlSystems'
+
+# debug codes - these happen occasionally, but the cause is unknown
+EVO_DEBUG_NO_RECENT_UPDATES = '0x01'
+EVO_DEBUG_NO_STATUS = '0x02'
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Create a Honeywell (EMEA/EU) evohome CH/DHW system.
+
+    An evohome system consists of: a controller, with 0-12 heating zones (e.g.
+    TRVs, relays) and, optionally, a DHW controller (a HW boiler).
+
+    Here, we add the controller, and the zones (if there are any).
+    """
+    domain_data = hass.data[DATA_EVOHOME]
+
+    client = domain_data['client']
+    loc_idx = domain_data['params'][CONF_LOCATION_IDX]
+
+    # evohomeclient has no defined way of accessing non-default location other
+    # than using a protected member, such as below
+    tcs_obj_ref = client.locations[loc_idx]._gateways[0]._control_systems[0]    # noqa E501; pylint: disable=protected-access
+
+    _LOGGER.debug(
+        "setup_platform(): Found Controller [idx=%s]: id: %s [%s], type: %s",
+        loc_idx,
+        tcs_obj_ref.systemId,
+        tcs_obj_ref.location.name,
+        tcs_obj_ref.modelType
+    )
+    parent = EvoController(domain_data, client, tcs_obj_ref)
+    add_entities([parent], update_before_add=True)
+
+
+class EvoController(ClimateDevice):
+    """Base for a Honeywell evohome hub/Controller device.
+
+    The Controller (aka TCS, temperature control system) is the parent of all
+    the child (CH/DHW) devices.
+    """
+
+    def __init__(self, domain_data, client, obj_ref):
+        """Initialize the evohome entity.
+
+        Most read-only properties are set here.  So are pseudo read-only,
+        for example name (which _could_ change between update()s).
+        """
+        self.client = client
+        self._obj = obj_ref
+
+        self._id = obj_ref.systemId
+        self._name = domain_data['config']['locationInfo']['name']
+
+        self._config = domain_data['config'][GWS][0][TCS][0]
+        self._params = domain_data['params']
+
+        self._status = {}
+        self._timers = domain_data['timers']
+        self._timers['statusUpdated'] = datetime.min
+        domain_data['schedules'] = {}
+
+        self._available = False  # should become True after first update()
+
+        if _LOGGER.isEnabledFor(logging.DEBUG):
```

Comment:
Remove this logging.

Reply:
I want to log the configuration somewhere.  This is done once only, what is wrong with that?

Where else can I put it?

Reply:
If it's really necessary move it to setup_platform where there already is a debug log.

Reply:
Ok, I took it out if `__init__()`, simplified logging in `setup_platform()`, and expanded logging in `setup()` a little.